service/schema/SchemaService.kt

```diff
@@ -266,6 +267,7 @@ class SchemaService(
             coroutines.launchAndHandle(notification, LOGGER) {
                 openOrGetWriteTx()?.let { tx ->
                     function(tx)
+                    loadedTypeState.reset()
```

Comment:
We reset the loaded type state in three instances but with one guiding principle; have we opened a new transaction?
1. When a new write transaction is opened.
2. When a new read transaction is opened.
3. When we've performed a write transaction ourselves.